search.go

```diff
@@ -0,0 +1,76 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"regexp"
+	"strings"
+)
+
+const (
+	golangBinariesURL = "https://api.ftp-master.debian.org/binary/by_metadata/Go-Import-Path"
+)
+
+func getGolangBinaries() (map[string]string, error) {
+	golangBinaries := make(map[string]string)
+
+	resp, err := http.Get(golangBinariesURL)
+	if err != nil {
+		return nil, fmt.Errorf("getting %q: %v", golangBinariesURL)
+	}
+	if got, want := resp.StatusCode, http.StatusOK; got != want {
+		return nil, fmt.Errorf("unexpected HTTP status code: got %d, want %d", got, want)
+	}
+	var pkgs []struct {
+		Binary     string `json:"binary"`
+		ImportPath string `json:"metadata_value"`
+		Source     string `json:"source"`
+	}
+	if err := json.NewDecoder(resp.Body).Decode(&pkgs); err != nil {
+		return nil, err
+	}
+	for _, pkg := range pkgs {
+		if !strings.HasSuffix(pkg.Binary, "-dev") {
+			continue // skip -dbgsym packages etc.
+		}
+		golangBinaries[pkg.ImportPath] = pkg.Binary
+	}
+	return golangBinaries, nil
+}
+
+func execSearch(args []string) {
+	fs := flag.NewFlagSet("search", flag.ExitOnError)
+
+	err := fs.Parse(args)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if fs.NArg() != 1 {
+		fmt.Fprintf(os.Stderr, "Usage: %s search <pattern>\n", os.Args[0])
+		fmt.Fprintf(os.Stderr, "Uses Go's default regexp syntax\n")
+		fmt.Fprintf(os.Stderr, "Example: %s search debian\n", os.Args[0])
```

Comment:
Could we make the example use a common regexp so that people can easily adapt the example without having to consult the whole documentation? How about “debi.*” for example?

Reply:
Alright! I have switched to your example!